src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerfileFunctionalTest.groovy

```diff
@@ -14,8 +14,11 @@ class DockerfileFunctionalTest extends AbstractFunctionalTest {
         File source = new File(TestConfiguration.class.getClassLoader().getResource("Dockerfile.template").toURI())
         if (source.exists()) {
             File resourcesDir = new File(projectDir, 'src/main/docker/')
-            resourcesDir.mkdirs()
-            Files.copy(source.toPath(),Paths.get(projectDir.path, 'src/main/docker/Dockerfile.template'))
+            if (resourcesDir.mkdirs()) {
+                Files.copy(source.toPath(),Paths.get(projectDir.path, 'src/main/docker/Dockerfile.template'))
```

Comment:
Dealing with files can be a pain but lets check to ensure that data was actually written:

```
if (Files.copy(source.toPath(),Paths.get(projectDir.path, 'src/main/docker/Dockerfile.template')) != source.length()) {
    throw new GradleException("File could not be successfully copied")
} 
```